Sources/Compiler/AST/NodeIDs/NodeKind.swift

```diff
@@ -1,315 +1,196 @@
-    case .integerLiteralExpr        : return "IntegerLiteralExpr"
-    case .lambdaExpr                : return "LambdaExpr"
-    case .mapLiteralExpr            : return "MapLiteralExpr"
-    case .matchExpr                 : return "MatchExpr"
-    case .nameExpr                  : return "NameExpr"
-    case .nilExpr                   : return "NilExpr"
-    case .sequenceExpr              : return "SequenceExpr"
-    case .storedProjectionExpr      : return "StoredProjectionExpr"
-    case .stringLiteralExpr         : return "StringLiteralExpr"
-    case .subscriptCallExpr         : return "SubscriptCallExpr"
-    case .tupleExpr                 : return "TupleExpr"
-    case .tupleMemberExpr           : return "TupleMemberExpr"
-    case .unicodeScalarLiteralExpr  : return "UnicodeScalarLiteralExpr"
-
-    case .pattern                   : return "Pattern"
-    case .bindingPattern            : return "BindingPattern"
-    case .exprPattern               : return "ExprPattern"
-    case .namePattern               : return "NamePattern"
-    case .tuplePattern              : return "TuplePattern"
-    case .wildcardPattern           : return "WildcardPattern"
-
-    case .stmt                      : return "Stmt"
-    case .braceStmt                 : return "BraceStmt"
-    case .breakStmt                 : return "BreakStmt"
-    case .continueStmt              : return "ContinueStmt"
-    case .declStmt                  : return "DeclStmt"
-    case .doWhileStmt               : return "DoWhileStmt"
-    case .exprStmt                  : return "ExprStmt"
-    case .forStmt                   : return "ForStmt"
-    case .returnStmt                : return "ReturnStmt"
-    case .whileStmt                 : return "WhileStmt"
-    case .yieldStmt                 : return "YieldStmt"
-
-    case .typeExpr                  : return "TypeExpr"
-    case .asyncTypeExpr             : return "AsyncTypeExpr"
-    case .conformanceLensTypeExpr   : return "ConformanceLensTypeExpr"
-    case .existentialTypeExpr       : return "ExistentialTypeExpr"
-    case .indirectTypeExpr          : return "IndirectTypeExpr"
-    case .lambdaTypeExpr            : return "LambdaTypeExpr"
-    case .nameTypeExpr              : return "NameTypeExpr"
-    case .parameterTypeExpr         : return "ParameterTypeExpr"
-    case .storedProjectionTypeExpr  : return "StoredProjectionTypeExpr"
-    case .tupleTypeExpr             : return "TupleTypeExpr"
-    case .unionTypeExpr             : return "UnionTypeExpr"
-    case .wildcardTypeExpr          : return "WildcardTypeExpr"
-
-    case .matchCase                 : return "MatchCase"
-    case .topLevelDeclSet           : return "TopLevelDeclSet"
-
-    default                         : return("Unknown")
-    }
-  }
+    MatchCase.self,
+    TopLevelDeclSet.self
+  ]
 
+  static let indices = Dictionary(
+    uniqueKeysWithValues: allValues.enumerated().lazy.map { (n, t) in (NodeKind(t), n) })
```

Comment:
```suggestion
    uniqueKeysWithValues: allValues.enumerated().map { (n, t) in (NodeKind(t), n) })
```

`lazy` is not as efficient as you seem to think. In fact, I believe it makes things slower unless the collection is really large.

I profiled this line; it's faster without the laziness.

Reply:
```suggestion
    uniqueKeysWithValues: allValues.enumerated().map { (n, t) in (NodeKind(t), n) })
```